config.go

```diff
@@ -99,26 +99,34 @@ func toMapMode(mode string) int {
 }
 
 type projConfig struct {
-	Zorder              []interface{}    `json:"zorder"`
-	Scenes              []*costumeConfig `json:"scenes"`
-	Costumes            []*costumeConfig `json:"costumes"`
-	CurrentCostumeIndex *int             `json:"currentCostumeIndex"`
-	SceneIndex          int              `json:"sceneIndex"`
+	Zorder        []interface{}     `json:"zorder"`
+	Backdrops     []*backdropConfig `json:"backdrops"`
+	BackdropIndex int               `json:"backdropIndex"`
+	Map           mapConfig         `json:"map"`
+	Camera        *cameraConfig     `json:"camera"`
+	Run           *Config           `json:"run"`
 
-	Map    mapConfig     `json:"map"`
-	Camera *cameraConfig `json:"camera"`
-
-	Run *Config `json:"run"`
+	// deprecated properties
+	Scenes              []*backdropConfig `json:"scenes"`              //this property is deprecated, use Backdrops instead
+	Costumes            []*backdropConfig `json:"costumes"`            //this property is deprecated, use Backdrops instead
+	CurrentCostumeIndex *int              `json:"currentCostumeIndex"` //this property is deprecated, use BackdropIndex instead
+	SceneIndex          int               `json:"sceneIndex"`          //this property is deprecated, use BackdropIndex instead
 }
 
-func (p *projConfig) getScenes() []*costumeConfig {
+func (p *projConfig) getBackdrops() []*backdropConfig {
 	if p.Scenes != nil {
 		return p.Scenes
 	}
-	return p.Costumes
+	if p.Costumes != nil {
+		return p.Costumes
+	}
+	return p.Backdrops
 }
 
-func (p *projConfig) getSceneIndex() int {
+func (p *projConfig) getBackdropIndex() int {
+	if p.BackdropIndex != 0 {
```

Comment:
It seems `p.BackdropIndex == 0` is a normal case, so it can't be a non-exist flag.

Reply:
The logic is actually not problematic
If it's a configuration for an older version, the BackdropIndex should == 0. 
If the BackdropIndex != 0, it indicates that the user is using the new standard, and at this point, it should return BackdropIndex

Reply:
You're right. When using a new editor to modify old files, it can indeed introduce subtle bugs.( The editor changes the new values to 0, but doesn't remove the old fields,  causing the new values ineffective)